bundles/org.openhab.binding.luxtronikheatpump/src/main/java/org/openhab/binding/luxtronikheatpump/internal/HeatpumpConnector.java

```diff
@@ -0,0 +1,195 @@
+            dataout.close();
+        }
+    }
+
+    /**
+     * read the parameters of the heat pump
+     */
+    public Integer[] getParams() {
+        return heatpumpParams;
+    }
+
+    /**
+     * set a parameter of the heat pump
+     * 
+     * @param param
+     * @param value
+     * @throws IOException indicate that no data can be sent to the heat pump
+     */
+    public Boolean setParam(int param, int value) throws IOException {
+        try (Socket sock = new Socket(serverIp, serverPort)) {
+            InputStream in = sock.getInputStream();
+            OutputStream out = sock.getOutputStream();
+            DataInputStream datain = new DataInputStream(in);
+            DataOutputStream dataout = new DataOutputStream(out);
+
+            while (datain.available() > 0) {
+                datain.readByte();
+            }
+            dataout.writeInt(SOCKET_PARAM_WRITE_PARAMS);
+            dataout.writeInt(param);
+            dataout.writeInt(value);
+            dataout.flush();
+
+            int cmd = datain.readInt();
+            datain.readInt();
+
+            datain.close();
+            dataout.close();
+
+            if (cmd != SOCKET_PARAM_WRITE_PARAMS) {
+                logger.debug("Couldn't write parameter {} with value {} to heat pump.", param, value);
+                return false;
+            } else {
+                logger.debug("Parameter {} with value {} successfully written to heat pump.", param, value);
+                return true;
+            }
+        }
+    }
+
+    /**
+     * Returns the internal states of the heat pump
+     * 
+     * @return a array with all internal data of the heat pump
+     */
+    public Integer[] getValues() {
+        return heatpumpValues;
+    }
+
+    /**
+     * Returns the internal visibilities of the heat pump
+     *
+     * @return a array with all internal visibilities of the heat pump
+     */
+    public Integer[] getVisibilities() {
+        return heatpumpVisibilities;
+    }
+
+    /**
+     * Reads all available parameters for the given value from socket
+     * 
+     * @param datain data input stream of socket connection
+     * @param dataout data output stream of socket connection
+     * @param value int value to read from socket
+     * @return an array with all values returned from heat pump socket
+     * @throws IOException indicate that no data can be read from the heat pump
+     */
+    private Integer[] readInt(DataInputStream datain, DataOutputStream dataout, int value) throws IOException {
+        Integer[] result = null;
+        while (datain.available() > 0) {
+            datain.readByte();
+        }
+        dataout.writeInt(value);
+        dataout.writeInt(0);
+        dataout.flush();
+        if (datain.readInt() != value) {
+            return new Integer[0];
+        }
+
+        if (value == SOCKET_PARAM_READ_VALUES) {
+            datain.readInt();
+        }
+
+        int arraylength = datain.readInt();
```

Comment:
Just a thought: Maybe an upper limit would be useful as int's max could lead to try to allocate ~8G RAM (assuming an `Integer`-Pointer is 4 Bytes because of an 32-Bit system)

Reply:
You mean in case the socket returns (for any reason) a number that is a lot higher than expected. Shouldn't happen I think. But won't hurt to add a check